### PR TITLE
fixed: truebeta refinement was broken

### PR DESCRIPTION
**Pull request by akva2**
need to remap to elements, not basis functions

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-stokes=79 ifem-openfrac=19 please

File: src/ASM/ASMunstruct.h [hunk 1]
```diff
@@ -168,7 +168,9 @@ class ASMunstruct : public ASMbase
   //! \brief Remaps element-wise errors from geometry mesh to refinement mesh.
   //! \param[out] errors The remapped errors
   //! \param[in] orig The element-wise errors on the geometry mesh
-  virtual void remapErrors(RealArray& errors, const RealArray& orig) const = 0;
+  //! \param[in] elemErrors If true, map to elements instead of basis functions
+  virtual void remapErrors(RealArray& errors, const RealArray& orig,
+                           bool elemErrors) const = 0;
```
Comment (kmokstad) [inline src/ASM/ASMunstruct.h]: Maybe with default value = false such that the method might be invoked as before without change in behaviour

File: src/ASM/LR/ASMu2D.C [hunk 2]
```diff
@@ -2180,13 +2180,17 @@ bool ASMu2D::matchNeighbour (ASMunstruct* neigh, int midx, int sidx, int orient)
...
 
-  for (const LR::Element* elm : basis->getAllElements())
-    for (const LR::Basisfunction* b : elm->support())
-      errors[b->getId()] += origErr[elm->getId()];
+  if (elemErrors)
+    errors = origErr;
+  else
+    for (const LR::Element* elm : basis->getAllElements())
+      for (const LR::Basisfunction* b : elm->support())
+        errors[b->getId()] += origErr[elm->getId()];
```
Comment (kmokstad) [inline src/ASM/LR/ASMu2D.C]: Better with
```
if (elemErrors) {
  errors = origErr;
  return;
}
```
before line 2186 and the rest can be unchanged (less diff)

File: src/ASM/LR/ASMu3D.C [hunk 3]
```diff
@@ -2250,13 +2250,17 @@ bool ASMu3D::matchNeighbour(ASMunstruct* neigh, int midx, int sidx, int orient)
...
 
-  for (const LR::Element* elm : basis->getAllElements())
-    for (const LR::Basisfunction* b : elm->support())
-      errors[b->getId()] += origErr[elm->getId()];
+  if (elemErrors)
+    errors = origErr;
+  else
+    for (const LR::Element* elm : basis->getAllElements())
+      for (const LR::Basisfunction* b : elm->support())
+        errors[b->getId()] += origErr[elm->getId()];
```
Comment (kmokstad) [inline src/ASM/LR/ASMu3D.C]: same comment here as for the ASMu2D method

File: src/SIM/AdaptiveSIM.C [hunk 4]
```diff
@@ -364,7 +365,7 @@ bool AdaptiveSIM::adaptMesh (int iStep)
 
       // remap from geometry basis to refinement basis
       Vector locErr(patch->getNoNodes(rBasis));
-      static_cast<ASMunstruct*>(patch)->remapErrors(locErr, locNorm);
+      static_cast<ASMunstruct*>(patch)->remapErrors(locErr, locNorm, false);
```
Comment (kmokstad) [inline src/SIM/AdaptiveSIM.C]: not needed, if the elemeErrors argument has default value = false, as suggested

**Comment by akva2** [comment]
all suggestions taken, merging.

File: src/ASM/LR/ASMu2D.C [hunk 5]
```diff
@@ -2180,10 +2180,16 @@ bool ASMu2D::matchNeighbour (ASMunstruct* neigh, int midx, int sidx, int orient)
...
+void ASMu2D::remapErrors (RealArray& errors,
+                          const RealArray& origErr, bool elemErrors) const
 {
   const LR::LRSplineSurface* basis = this->getBasis(1);
 
+  if (elemErrors) {
+    errors = origErr;
+    return;
+  }
+
```
Comment (kmokstad) [inline src/ASM/LR/ASMu2D.C]: I wrote _before_ line 2186 (now the basis variable is unused of elemErrors is true, but never mind. Some other time (also for the ASUm3D)

Comment (akva2) [reply src/ASM/LR/ASMu2D.C]: woops. missed that bit. you are ofc right.

**Pull request merged**